[PropertyInfo] Fix dock block lookup fallback loop

DerManoMann opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

`getDocBlock()` in `PhpDocExtractor` implements a fallback loop when trying to lookup a dock block for a property (property, accessor method, mutator method). This relies on the individual lookups to return `null`.

Unfortunately, phpDocumentor will throw an `InvalidArgumentException` exception in case there is no dock block.

Currently the try/catch is wrapping the whole loop which means if the first lookup fails the loop is ended instead of moving on to the next one.

This PR moves the try/catch to each individual call of `$this->docBlockFactory->create()`.

DerManoMann committed:
[PropertyInfo] Fix dock block lookup fallback loop

nicolas-grekas commented:
Thanks for the PR and the clear description. Could you add a test case please?

DerManoMann commented:
Ah, looks like i rebased on master, not 4.1 ... is there an easy fix for this or is it better to create a new PR, I wonder?

dunglas commented on src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php:
```diff
@@ -194,7 +190,13 @@ private function getDocBlockFromProperty(string $class, string $property): ?DocB
+            // No valid doc block
+        }
+
+        return null;
```
You can return in the catch (same below)

DerManoMann replied:
Was wondering ... I guess that a personal preference thing for me. I always feel it looks weird to not have a return at the end if the return type is not void.

vudaltsov committed:
AppBundle->App.

nicolas-grekas committed:
minor #27869 Replaced AppBundle mentions with App (vudaltsov)

This PR was merged into the 3.4 branch.

Discussion
----------

Replaced AppBundle mentions with App

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |

Removed some legacy non-test related mentions of AppBundle. Probably this should be done for 4.x only, not >=3.4.

Commits
-------

b4552373c1 AppBundle->App.

nicolas-grekas commented:
Thank you @DerManoMann.

nicolas-grekas commented:
Merged on branch 3.4 by the way.
